Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1

```diff
@@ -0,0 +1,795 @@
+﻿<###
+# Overview:
+    This script is used to configure WAF Policy on all endpoints of Front Door CDN(s) in a Subscription.
+
+# Control ID:
+    Azure_FrontDoor_CDNProfile_NetSec_Enable_WAF_Configuration
+
+# Display Name:
+   Front Door CDN should have Web Application Firewall configured
+
+# Prerequisites:
+    1. Contributor or higher privileges on the Front Door CDNs in a Subscription.
+    2. Must be connected to Azure with an authenticated account.
+
+# Steps performed by the script:
+    To remediate:
+        1. Validate and install the modules required to run the script.
+        2. Get the list of all Front Door CDNs Endpoints in a Subscription that do not have WAF Policy Configured
+        3. Back up details of Front Door CDN Endpoint(s) that are to be remediated.
+        4. Configure WAF for all endpoints in the Front Door.
+
+    To roll back:
+        1. Validate and install the modules required to run the script.
```

Comment:
Do we validate again in rollback?

Reply:
Yes

Reply:
Could we rollback without executing remediation command. Just thinking of scenarios where rollback could be run in isolation.

Reply:
we do roll back only for remediated resource only.